src/formats/farbfeld.zig

```diff
@@ -0,0 +1,119 @@
+    height: u32 align(1),
+
+    pub const size = 16;
+    const width_size = 4;
+    const height_size = 4;
+    const magic_value: [8]u8 = "farbfeld".*;
+
+    fn encode(header: Header) [size]u8 {
+        var result: [size]u8 = undefined;
+        @memcpy(result[0..8], &magic_value);
+        std.mem.writeInt(u32, result[8..12], header.width, .big);
+        std.mem.writeInt(u32, result[12..16], header.height, .big);
+        return result;
+    }
+    comptime {
+        assert(@sizeOf(Header) + @sizeOf(@TypeOf(magic_value)) == size);
+    }
+};
+
+pub const Farbfeld = struct {
+    header: Header,
+    pixels: []color.Rgba64,
+
+    pub fn formatInterface() FormatInterface {
+        return FormatInterface{
+            .format = format,
+            .formatDetect = formatDetect,
+            .readImage = readImage,
+            .writeImage = writeImage,
+        };
+    }
+
+    pub fn format() ImageUnmanaged.Format {
+        return ImageUnmanaged.format.farbfeld;
+    }
+    /// Taken a stream, Returns true if and only if the stream contains the magic value "fabfeld"
+    pub fn formatDetect(stream: *ImageUnmanaged.Stream) ReadError!bool {
+        var magic_buffer: [Header.magic_value.len]u8 = undefined;
+        const bytes_read = try stream.read(magic_buffer[0..]);
+        return bytes_read == Header.magic_value.len and
+            std.mem.eql(u8, magic_buffer[0..], Header.magic_value[0..]);
+    }
+    pub fn read(allocator: std.mem.Allocator, stream: *ImageUnmanaged.Stream) ReadError!Farbfeld {
+        var farbfeld: Farbfeld = undefined;
+        // read header magic value
+        var buffered_stream = buffered_stream_source.bufferedStreamSourceReader(stream);
+        const reader = buffered_stream.reader();
+
+        const magic_match: bool = reader.isBytes(&Header.magic_value) catch return ReadError.InvalidData;
+        if (!magic_match)
+            return ReadError.InvalidData;
+
+        // read width and height
+        const header = utils.readStruct(reader, Header, .big) catch return ReadError.InvalidData;
+        farbfeld.header = header;
+
+        farbfeld.pixels = try allocator.alloc(color.Rgba64, @as(usize, header.width) * @as(usize, header.height));
+        errdefer allocator.free(farbfeld.pixels);
+
+        for (farbfeld.pixels) |*pixel| {
+            const pixel_color = utils.readStruct(reader, color.Rgba64, .big) catch return ReadError.InvalidData;
+            pixel.* = pixel_color;
+        }
+        return farbfeld;
+    }
+    pub fn readImage(allocator: std.mem.Allocator, stream: *ImageUnmanaged.Stream) ReadError!ImageUnmanaged {
+        var image: ImageUnmanaged = .{};
+        const farbfeld = try read(allocator, stream);
+        image.width = farbfeld.header.width;
+        image.height = farbfeld.header.height;
+        image.pixels = .{ .rgba64 = farbfeld.pixels };
+        return image;
+    }
+    pub fn write(self: Farbfeld, write_stream: *ImageUnmanaged.Stream) WriteError!void {
```

Comment:
This should be:
`pub fn write(self: Ferbfeld, stream: *ImageUnamanged.Stream, pixels: color.PixelStorage) ImageUnamanged.WriteError!void`

If the user does not pass a PixeStorage of format rgba32, return an error.